compiler/codegen/OMRCodeGenerator.cpp

```diff
@@ -3055,6 +3066,16 @@ TR::Instruction *OMR::CodeGenerator::generateDebugCounter(TR::Instruction *curso
    TR::DebugCounter *counter = TR::DebugCounter::getDebugCounter(self()->comp(), name, fidelity, staticDelta);
    if (!counter)
       return cursor;
+
+   if (self()->comp()->compileRelocatableCode() && !counter->getReloData(self()->comp(), 0, fidelity, staticDelta))
+      return cursor;
```

Comment:
Just want to confirm that this case means: we're compiling relocatable code but don't know how to relocate this particular requested debug counter, so it silently just doesn't insert the counter?

Reply:
What it actually means is "we failed to allocate memory we need to store information about this counter at relo time". I chose the "silently fail to insert counter" path because of the previous check, which is also effectively a check to see if we managed to allocate the counter.